File: Deepseek-Coder-33b-Instruct/stage2/serial_construct_wait_clause.c
#include <stdio.h>
#include <stdlib.h>
#include <openacc.h>

#ifndef T1
//T1:serial construct wait clause,V:2.7-3.3
int test1(){
    int err = 0;
    int data[100];
    int i;

    // Initialize data
    for (i = 0; i < 100; i++) {
        data[i] = i;
    }

    // Parallel region
    #pragma acc parallel loop
    for (i = 0; i < 100; i++) {
        data[i] *= 2;
    }

    // Wait for all kernels to finish
    #pragma acc wait

    // Check the results
    for (i = 0; i < 100; i++) {
        if (data[i] != i * 2) {
            err = 1;
            break;
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    failed = failed + test1();
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/serial_construct_wait_clause.f90
#ifndef T1
!T1:serial construct wait clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 100
        INTEGER :: i
        REAL(8), DIMENSION(N) :: a, b
        REAL(8) :: sum

        ! Initialize arrays
        DO i = 1, N
          a(i) = i
          b(i) = i * 2
        END DO

        ! Compute sum on the host
        sum = 0.0
        !$acc serial
        DO i = 1, N
          sum = sum + a(i) + b(i)
        END DO
        !$acc end serial

        ! Check the result
        IF (sum .ne. N * (N + 1)) THEN
          errors = errors + 1
        END IF

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM